lib/src/main/java/com/futurewei/alcor/common/db/ignite/IgniteClientDbCache.java

```diff
@@ -223,4 +283,209 @@ public long size() {
     public Transaction getTransaction() {
         return transaction;
     }
+
+    private boolean checkForSqlFieldsQuery(Map<String, Object[]> filterParams) {
+        if (!checkedForSqlFields)
+            return false;
+        if (sqlFields == null || sqlFields.size() == 0 || filterParams.size() == 0)
+            return false;
+
+        /*
+        * All filterParams must be found in sqlfields as "index".
+         */
+        int idxCount = 0;
+        for (String f : filterParams.keySet()) {
+            if (sqlFields.containsKey(f))
+                ++idxCount;
+        }
+
+        return idxCount != 0 && idxCount == filterParams.size();
+    }
+
+    @DurationStatistics
+    private <E1, E2> V getBySqlFields(Map<String, Object[]> filterParams) throws CacheException {
+        try {
+            Map<K, V> result = runSQLFieldsQuery(filterParams);
+            if (result == null || result.isEmpty())
+                return null;
+            if(result.size() > 1) {
+                throw new CacheException(NON_SCALAR_ROWSET);
+            }
+            return result.get(0);
+        }
+        catch (CacheException ce) {
+            if (ce.getMessage().equals(NON_SCALAR_ROWSET))
+                throw ce;
+        }
+        catch (Exception e) {
+            logger.log(Level.WARNING, "runSQLFieldsQuery exception : " + e.getMessage());
+        }
+
+        return null;
+    }
+
+    @DurationStatistics
+    public <E1, E2> Map<K, V> getBySqlFieldsAll(Map<String, Object[]> filterParams) {
```

Comment:
@pkommoju Do we need to throw exception from this function?

Reply:
I think there is no need. runSqlFieldsQuery will throw any exception anyway. Just need to add throws CacheException.